### PR TITLE
Adding root check on default control socket

### PR DESCRIPTION
**Pull request by fpoussin**
This removes the need to use --system when dinitctl runs as root.

### CODE REVIEW

**Comment by davmac314** [comment]
Other than above, the documentation should be updated to reflect this change (eg `README.md` refers to using the "-s" switch to talk to the system instance, which shouldn't be necessary now, and the examples should be updated; man page for dinitctl (in `doc/manpages/dinitctl.8.m4`) also needs updating to say that "-s" is the default if run as root.

If you don't want to do the documentation changes, I'm happy to do them myself (when I get the time), but I won't merge this change until they are ready (on the master branch, I want to keep documentation consistent with behaviour).

Thanks!

**Comment by fpoussin** [comment]
Alright, I'll make the changes.

File: src/dinitctl.cc [hunk 1]
```diff
@@ -294,7 +294,7 @@ int main(int argc, char **argv)
     }
     else {
         control_socket_path = SYSCONTROLSOCKET; // default to system
-        if (! sys_dinit) {
+        if (! sys_dinit && getuid() != 0) { // no --system flag and not root
```
[the remark this answers is not in the document]
Comment (fpoussin) [reply src/dinitctl.cc]: Good idea!

**Comment by fpoussin** [comment]
There you go.
The default is now to use system daemon as root, user daemon as user.
I've replaced the system flag with a user flag as it makes more sense than adding it.
What do you think?

**Comment by davmac314** [comment]
This looks good, thanks! (I might restore the "-s" option before next release, to retain backwards compatibility, but it's fine to remove it for now).

If you would like to, you can add yourself as a contributor in the CONTRIBUTORS file. Please open another pull request for that if you do want that. #

**Pull request merged**